docs/reference/api/widgets/table.rst

```diff
@@ -1,55 +1,112 @@
 Table
 =====
 
+A widget for displaying columns of tabular data.
+
+.. figure:: /reference/images/Table.png
+   :width: 300px
+   :align: center
+
 .. rst-class:: widget-support
 .. csv-filter:: Availability (:ref:`Key <api-status-key>`)
    :header-rows: 1
    :file: ../../data/widgets_by_platform.csv
    :included_cols: 4,5,6,7,8,9,10
    :exclude: {0: '(?!^(Table|Component)$)'}
 
-The table widget is a widget for displaying tabular data. It can be instantiated with the list of headings and then data rows
-can be added.
-
-.. figure:: /reference/images/Table1.jpeg
-    :align: center
-
 Usage
 -----
 
+A Table uses a :class:`~toga.sources.ListSource` to manage the data being displayed.
+options. If ``data`` is not specified as a ListSource, it will be converted into a
+ListSource at runtime.
+
+The simplest instantiation of a Table is to use a list of lists (or list of tuples),
+containing the items to display in the table. When creating the table, you must also
+specify the headings to use on the table; those headings will be converted into
+accessors on the Row data objects created for the table data. In this example,
+we will display a table of 2 columns, with 3 initial rows of data:
+
+.. code-block:: python
+
+    import toga
+
+    table = toga.Table(
+        headings=["Name", "Age"],
+        data=[
+            ("Arthur Dent", 42),
+            ("Ford Prefect", 37),
+            ("Tricia McMillan", 38),
+        ]
+    )
+
+    # Get the details of the first item in the data:
+    print(f"{table.data[0].name} is age {table.data[0].age}")
+
+    # Append new data to the table
+    table.data.append(("Zaphod Beeblebrox", 47))
+
+You can also specify data for a Table using a list of dictionaries. This allows you to
+store data in the data source that won't be displayed in the table. It also allows you
+to control the display order of columns independent of the storage of that data.
+
 .. code-block:: python
 
     import toga
 
-    table = toga.Table(['Heading 1', 'Heading 2'])
+    table = toga.Table(
+        headings=["Name", "Age"],
+        data=[
+            {"name": "Arthur Dent", "age": 42, "planet": "Earth"},
+            {"name", "Ford Prefect", "age": 37, "planet": "Betelgeuse Five"},
+            {"name": "Tricia McMillan", "age": 38, "plaent": "Earth"},
+        ]
+    )
+
+    # Get the details of the first item in the data:
+    print(f"{table.data[0].name}, who is age {table.data[0].age}, "
+          f"is from {table.data[0].planet}")
```

Comment:
Our example code should be black-compatible:
```suggestion
    print(
        f"{table.data[0].name}, who is age {table.data[0].age}, "
        f"is from {table.data[0].planet}"
    )
```

Reply:
Even better, I've factored out `table.data[0]` into a variable so it doesn't even need to wrap anymore.